store/storage/pebbledb/db.go

```diff
@@ -21,17 +21,24 @@ const (
 	tombstoneVal     = "TOMBSTONE"
 )
 
-var (
-	_ store.VersionedDatabase = (*Database)(nil)
-
-	defaultWriteOpts = pebble.Sync
-)
+var _ store.VersionedDatabase = (*Database)(nil)
 
 type Database struct {
 	storage *pebble.DB
+
+	// Sync is whether to sync writes through the OS buffer cache and down onto
+	// the actual disk, if applicable. Setting Sync is required for durability of
+	// individual write operations but can result in slower writes.
+	//
+	// If false, and the process or machine crashes, then a recent write may be
+	// lost. This is due to the recently written data being buffered inside the
+	// process running Pebble. This differs from the semantics of a write system
+	// call in which the data is buffered in the OS buffer cache and would thus
+	// survive a process crash.
+	sync bool
 }
 
-func New(dataDir string) (*Database, error) {
+func New(dataDir string, sync bool) (*Database, error) {
```

Comment:
could we pass in options here instead or create a new function for creating pebble db with options?

Reply:
I don't seem to follow. WDYM by pass in options?

Reply:
New() would use the default options, then there would be a NewWithOptions `NewWithOptions(dir string, opts pebble.Options)`

Reply:
Ok, I'll keep `NewWithDB` as is (allowing a user to provide the storage and sync value directly), whereas `New` will just use the default 👍